src/visualization/server/plot.cpp

```diff
@@ -21,14 +22,32 @@
 
 namespace turi{
   namespace visualization{
-    Plot::Plot() {
-      // make UUID for plot
-      static auto uuid_generator = boost::uuids::random_generator();
-      auto uuid = uuid_generator();
-      m_id = boost::lexical_cast<std::string>(uuid);
+
+    Plot::Plot() { }
+
+    Plot::Plot(const std::string& vega_spec) {
+      // This constructor removes newlines from the vega spec passed in.
+      // In order to delineate messages between frontend and backend,
+      // the newline character is used as a separator.
+      m_vega_spec = make_format_string(vega_spec);
     }
 
+    Plot::Plot(const std::string& vega_spec, std::shared_ptr<transformation_base> transformer, double size_array) :
+      m_vega_spec(vega_spec),
+      m_size_array(size_array),
+      m_transformer(transformer) {
+        // This constructor expects builtin TC format strings
+        // that already have their newlines removed.
+        DASSERT_EQ(m_vega_spec.find("\n"), std::string::npos);
+      }
+
     const std::string& Plot::get_id() const {
+      static auto uuid_generator = boost::uuids::random_generator();
```

Comment:
Why not put this in the if?

Reply:
Since it's static, it's only going to run once (on the first call to this function). I think the semantics of static initialization are clearer if it's not inside an `if`, and we're going to use it on first call to this function anyway.